src/main/kotlin/omok/model/rule/OmokChecker.kt

```diff
@@ -0,0 +1,57 @@
+package omok.model.rule
+
+import omok.model.board.Board
+import omok.model.position.DeltaPosition
+import omok.model.position.Position
+import omok.model.stone.Stone
+
+object OmokChecker {
+    private const val DEFAULT_STONE_COUNT = 1
+    private const val OMOK_PRECONDITION = 5
+    private val horizontalDirection = DeltaPosition(0, 1)
+    private val verticalDirection = DeltaPosition(1, 0)
+    private val upwardDirection = DeltaPosition(1, 1)
+    private val downwardDirection = DeltaPosition(1, -1)
+
+    fun findOmok(
+        lastPosition: Position,
+        stone: Stone,
+    ): Boolean {
+        val directionType = listOf(horizontalDirection, verticalDirection, upwardDirection, downwardDirection)
+        return directionType.any { direction ->
+            checkWinningPosition(lastPosition, direction, stone)
+        }
+    }
+
+    private fun checkWinningPosition(
+        lastPosition: Position,
+        deltaPosition: DeltaPosition,
+        stone: Stone,
+    ): Boolean {
+        var stoneCount = DEFAULT_STONE_COUNT
+
+        stoneCount += countSameStones(lastPosition, deltaPosition, stone)
+
+        stoneCount += countSameStones(lastPosition, -deltaPosition, stone)
+
+        return stoneCount >= OMOK_PRECONDITION
+    }
```

Comment:
```suggestion
    private fun checkWinningPosition(
        lastPosition: Position,
        deltaPosition: DeltaPosition,
        stone: Stone,
    ): Boolean = (
            DEFAULT_STONE_COUNT
            + countSameStones(lastPosition, deltaPosition, stone)
            + countSameStones(lastPosition, -deltaPosition, stone)
    ) >= OMOK_PRECONDITION
```

가령 저의 취향인 expression 위주로 표현하면 이렇게 구현을 할 수도 있습니다.
저는 이렇게 하면 var 을 사용하지 않고 불변값으로만 표현을 할 수 있어서 좋아합니다.
다만 이건 정말 취향의 영역이므로 꼭 반영해주시지 않아도 괜찮습니다.